### PR TITLE
Fix finalized block root index

### LINKED ISSUE #3947: Finalized block index is missing blocks in finalized epoch
When we index finalized block roots, we are assuming that the checkpoint root is the starting point for walking up the ancestry tree to build the canonical finalized chain. However, the finalized root in the checkpoint is the highest block up to or including the block at the finalized slot. This leaves the other blocks in the epoch un-indexed and therefore return false when asking `db.IsFinalizedBlock(...)`. 

The solution here is to index these epoch blocks as finalized, but rebuild the end of the index every time a new finalized checkpoint is added. 

I might also recommend that we annotate or rename `db.IsFinalizedBlock(...)` to indicate that the block is _finalized and canonical_. 

Until this is resolved, initial sync is broken, but I'll have a resolution later today / tonight.

### PR DESCRIPTION
**Pull request by prestonvanloon**
Resolves #3947 

- [x] Implement logic
- [x] Test at runtime locally
- [x] Add regression tests

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Fails at runtime currently. Server is still serving a gap in blocks. Will continue tomorrow.

File: beacon-chain/sync/initial-sync/round_robin.go [hunk 1]
```diff
@@ -73,6 +73,11 @@ func (s *InitialSync) roundRobinSync(genesis time.Time) error {
 			// Handle block large block ranges of skipped slots.
 			start += count * uint64(lastEmptyRequests*len(peers))
 
+			// Short circuit start far exceeding the highest finalized epoch in some infinite loop.
+			if start > helpers.StartSlot(highestFinalizedEpoch()+1) {
```
Comment (rauljordan) [inline beacon-chain/sync/initial-sync/round_robin.go]: Any way this case can be covered in a test? Seems very important

Comment (nisdas) [reply beacon-chain/sync/initial-sync/round_robin.go]: +1

Comment (prestonvanloon) [reply beacon-chain/sync/initial-sync/round_robin.go]: This scenario is not very well understood. Lets file and issue and capture the conditions when it happens in the next testnet.

**Pull request merged**